backend/ownerSizeTotalWorker.js

```diff
@@ -0,0 +1,261 @@
+// (meaning it's not self-healing), we mighr want in the future to also randomly
+// do a full computation on some contracts.
+
+const updatedSizeList = new Set()
+const updatedSizeMap = new Map()
+const cachedUltimateOwnerMap = new Map()
+
+// Super-fast 8-bit hash for base58 with low bucket size standard deviation
+//  (i.e., the 256 buckets get about the same number of elements)
+const fastBase58Hash = (cid: string) => {
+  const len = cid.length
+  const a = cid.codePointAt(len - 2) || 0
+  const b = cid.codePointAt(len - 1) || 0
+  return ((a * 19) + (b + 19)) & 0xFF
+}
+
+/**
+ * Appends a CID to its corresponding temporary index bucket in the database.
+ * This marks the CID as needing processing in the next delta calculation.
+ * @param cid The CID to add to the temporary index.
+ * @returns Promise resolving when the append operation completes.
+ */
+const addToTempIndex = (cid: string) => {
+  return appendToIndexFactory(`_private_pendingIdx_ownerTotalSize_${fastBase58Hash(cid)}`)(cid)
+}
+
+/**
+ * Removes multiple CIDs from their respective temporary index buckets.
+ * Groups CIDs by bucket for efficient batch removal.
+ * @param cids An array of CIDs to remove.
+ * @returns Promise resolving when all removal operations complete.
+ */
+const removeFromTempIndex = (cids: string[]) => {
+  // 1. Group CIDs by their hash bucket.
+  const cidsByBucket = cids.reduce((acc: Map<number, Set<string>>, cv: string) => {
+    const bucket = fastBase58Hash(cv)
+    const ownedResourcesSet = acc.get(bucket)
+    if (ownedResourcesSet) {
+      ownedResourcesSet.add(cv)
+    } else {
+      acc.set(bucket, new Set([cv]))
+    }
+    return acc
+  }, new Map())
+
+  // 2. Create removal promises for each bucket. Execute all removal
+  // operations concurrently.
+  return Promise.all([...cidsByBucket].map(([bucket, cids]) => {
+    return removeFromIndexFactory(`_private_pendingIdx_ownerTotalSize_${bucket}`)([...cids])
+  }))
+}
+
+/**
+ * Runs once on worker startup or initialization.
+ * It loads all CIDs from the persistent temporary index buckets into the
+ * in-memory `updatedSizeList`.
+ * These CIDs represent items needing a full recalculation (potentially from a
+ * previous unclean shutdown).
+ */
+sbp('okTurtles.eventQueue/queueEvent', readyQueueName, async () => {
+  // Iterate through all 256 possible bucket keys.
+  for (let i = 0; i < 256; i++) {
+    // Fetch the content of the bucket
+    // (bypassCache isn't really needed here as the cache should be empty, but
+    // it's added for clarity.)
+    const data = await sbp('chelonia.db/get', `_private_pendingIdx_ownerTotalSize_${i}`, { bypassCache: true })
+    if (data) {
+      // Split the string and add mark each CID as requiring full recalculation
+      data.split('\x00').forEach((cid) => {
+        updatedSizeList.add(cid)
+      })
+    }
+  }
+
+  console.log(`[ownerSizeTotalWorker] Loaded ${updatedSizeList.size} CIDs for full recalculation.`)
```

Comment:
Request, could you make this `.info` and move it inside of `computeSizeTask`?

Then also put another `console.info` at the very end of `computeSizeTask` that says something like:

```
console.log(`[ownerSizeTotalWorker] Computed size for ${updatedSizeList.size} CIDs in ${time} seconds.`)
```